opt4j-core/src/main/java/org/opt4j/core/genotype/DoubleGenotype.java

```diff
@@ -138,9 +129,9 @@ public void init(Random random, int n) {
 			getLowerBound(n - 1);
 			getUpperBound(n - 1);
 		} catch (IndexOutOfBoundsException outOfBoundException) {
-			String message = outOfBoundException.getMessage() == null ? "" : outOfBoundException.getMessage() + "\n";
-			message += "Can not initialize a genotype with " + n + " entries with the specified bounds";
-			throw new IndexOutOfBoundsException(message);
+			throw new IllegalArgumentException(
```

Comment:
I think we had exactly this issue with Felix. I originally wanted to throw the IllegalArgumentException and he thought that rethrowing the exception would be better